samples/cube.lobster

```diff
@@ -0,0 +1,111 @@
+// globe in front of the camera.
+//
+// The example illustrates creating a simple mesh and rendering a texture on
+// the mesh in various orientations.
+// The example showcases the render_to_texture utility, which allows us to
+// render a 2D drawing on the surfaces of a 3D object, using a frame buffer.
+//
+// Kris Kowal <kris@cixar.com>
+
+import vec
+import color
+import util3d
+
+fatal(gl_window("Lobster Cube", 515, 515))
+
+// Generate a mesh for a unit square.
+let square_corners = [ xy_0, xy_x, xy_1, xy_y, xy_0 ]
+let square_mesh = gl_new_poly(square_corners)
+
+// How to break glass when the texture coordinates do not match the mesh
+// vertices:
+// let square_positions = map(square_corners) v: xyz(v, 0)
+// let square_texcoords1 = square_corners
+// let square_indices = [0, 1, 2, 2, 3, 0]
+// let square_mesh = gl_new_mesh(
+//     "PT",
+//     square_positions,
+//     [], // square_colors,
+//     [], // square_normals,
+//     square_texcoords1,
+//     [], // square_textcoords2,
+//     square_indices
+// )
+
+let lobster_tex = gl_load_texture("data/textures/lobster.jpg")
+assert lobster_tex
+
+let detail = 256
+let face_tex = map([
+    color_purple,
+    color_olive,
+    color_teal,
+    color_dark_red,
+    color_dark_blue,
+    color_dark_green,
+]) color:
+    render_to_texture(nil, xy_1i * detail, false, nil, 0):
+        gl_set_shader("textured")
+        gl_set_primitive_texture(0, lobster_tex)
+        // The texture colors are multiplied by the color in context.
+        // Since our texture is white on black, we can change the white to a
+        // unique color for each face of the world.
+        gl_color(color)
+        gl_rect(xy_1 * detail, 0)
+
+def transform_to_face(i, f):
+    gl_translate(xyz_0):
+        if i == 0: // -x
+            gl_translate(xyz_x / -2)
+            gl_rotate_z(-xy_y)
+            gl_rotate_x(-xy_y)
+            gl_translate((xyz_x + xyz_y) / -2)
+        else: if i == 1: // -y
+            gl_translate(xyz_y / -2)
+            gl_rotate_x(-xy_y)
+            gl_translate((xyz_x + xyz_y) / -2)
+        else: if i == 2: // -z
+            gl_translate(xyz_z / -2)
+            gl_translate((xyz_x + xyz_y) / -2)
+        else: if i == 3: // +z
+            gl_translate(xyz_z / 2)
+            gl_rotate_y(-xy_x)
+            gl_translate((xyz_x + xyz_y) / -2)
+        else: if i == 4: // +y
+            gl_translate(xyz_y / 2)
+            gl_rotate_z(-xy_x)
+            gl_rotate_x(-xy_y)
+            gl_translate((xyz_x + xyz_y) / -2)
+        else: if i == 5: // +x
+            gl_translate(xyz_x / 2)
+            gl_rotate_z(xy_y)
+            gl_rotate_x(-xy_y)
+            gl_translate((xyz_x + xyz_y) / -2)
+        f()
+
+// This demo is able to use camera_FPS_view but uses a different control model
+// to move the camera about the origin at a fixed “elevation”.
+def camera_GPS_update(upkey, leftkey, downkey, rightkey, elevation:float, mousesens:float, keyspeed:float):
```

Comment:
there is some of this functionality already in `3dutil.lobster` that may be reusable?

Reply:
This is pretty different from camera_FPS_update, since it keeps the camera pointed at the center of the model at a fixed distance.

I did not see another function that duplicates this behavior, but I’ll look again.

Reply:
Ok, fair enough.